plugins/GSdx/Renderers/Common/GSTextureCache.cpp

```diff
@@ -2077,6 +2077,13 @@ GSTexture* GSTextureCache::Palette::GetPaletteGSTexture() {
 	return m_tex_palette;
 }
 
+void GSTextureCache::Palette::UploadClut() {
+	if (!m_tex_palette) {
+		m_tex_palette = m_renderer->m_dev->CreateTexture(256, 1);
+		m_tex_palette->Update(GSVector4i(0, 0, m_pal, 1), m_clut, m_pal * sizeof(m_clut[0]));
```

Comment:
Is it expected that UploadClut will upload only once ?

Reply:
Yes because the clut content for a given `Palette` is copied at object creation and never changed, so also the palette need only to be uploaded once (possibly lazy upload)

Reply:
In this case, UploadClut could have a better name. Create/Initialize (once) vs Upload (can be repeated). 

You can also reduce the size of the texture. 256 is too big for small m_pal.

Reply:
Ok for name. As for Tex size, IIRC I tried and creating smaller textures lead to various errors. Will test again.

Reply:
Yeah creating texture with 16 as width breaks the plugin (both dx and ogl), while for big pal (256) it is good (as hardcoded in there). However it would be an interesting enhancement to create smaller textures for this use case.